tests/data/test_data_catalogue.py

```diff
@@ -114,3 +116,42 @@ def test_datasets_are_same_unit(example_catalogue):
     # change first datasets unit
     example_catalogue.datasets[0].unit = "gt"
     assert not example_catalogue.datasets_are_same_unit()
+
+
+def test_data_catalogue_copy(example_catalogue_small):
+    example_catalogue_small.load_all_data()
+    example_catalogue_copy = example_catalogue_small.copy()
+    example_catalogue_copy.datasets[0].data.changes = [-4]
+    assert example_catalogue_copy.datasets[0].data.changes[0] == -4
+    assert example_catalogue_small.datasets[0].data.changes[0] != -4
+
+
+def test_average_timeseries_in_catalogue(example_catalogue_small):
+    example_catalogue_small.load_all_data()
+    example_catalogue_small.datasets.append(copy.deepcopy(example_catalogue_small.datasets[0]))
+
+    # since both timeseries are the same average should give back same as inputtimeseries
+    result_timeseries, _ = example_catalogue_small.average_timeseries_in_catalogue(remove_trend=False,
+                                                                                   add_trend_after_averaging=False)
+    assert np.array_equal(result_timeseries.data.changes, example_catalogue_small.datasets[0].data.changes)
+
+    # double the values for next test
```

Comment:
please split these up into individual unit tests. It should be obvious instantly when running them which are failing and which are passing, and the failing of one should not prevent the passing of another. This is only possible if they are written as separate tests.

Reply:
Done! This was just lazyness ;)